pages/channel/[id].tsx

```diff
@@ -0,0 +1,50 @@
+import {
+  useToken,
+  LiveKitRoom,
+  VideoConference,
+} from '@livekit/components-react';
+import { useUser } from '@supabase/auth-helpers-react';
+import { useRouter } from 'next/router';
+
+import {
+  Room,
+  RoomEvent,
+  RemoteParticipant,
+  RemoteTrackPublication,
+  RemoteTrack,
+  Participant,
+  VideoPresets,
+} from 'livekit-client';
+
+export default function Channel() {
+  const router = useRouter();
+  const params =
+    typeof window !== 'undefined' ? new URLSearchParams(location.search) : null;
+
+  const userIdentity = useUser();
+  console.log(userIdentity?.email);
+  console.log(userIdentity?.id);
+
+  const { id: roomRoute } = router.query;
+
+  const roomName: string = roomRoute as string;
+
+  const token = useToken(process.env.NEXT_PUBLIC_LK_TOKEN_ENDPOINT, roomName, {
```

Comment:
So main thing I'm wondering here is whether or not it's possible to use the existing token already generated by Supabase, which should already be part of the session object. If we can avoid maintaining multiple tokens it'd likely help to keep things somewhat cleaner overall.

Reply:
Slating this as future to combine tokens. Perhaps down the line we should ask the devs over @ LiveKit to support existing tokens or etc.